### PR TITLE
Feature task points

### PR DESCRIPTION
**Pull request by hannesi**
Functionality to fetch current task points from TIM

### CODE REVIEW

File: src/tidecli/utils/file_handler.py [hunk 1]
```diff
@@ -287,10 +289,13 @@ def get_task_file_data(file_path: Path, metadata: TaskData) -> list[TaskFile]:
...
 
-                        # Validator complains about answer.
-                        # Answer is submitted despite of complains.
+                    if validate_answer_file(answer_bycode, metadata_bycode):
+                        # TODO: tarvitaan lisää testitapauksia,
+                        logger.info("Gap-type exercise answer file is valid.")
                         f1.content = "\n".join(answer_gapcode)
-                        # return []
+                    else:
+                        logger.debug("Gap-type exercise answer not valid.")
```
Comment (sjsarsa) [inline src/tidecli/utils/file_handler.py:297]: missing file content update will cause wrong content to be sent

**Comment by sjsarsa** [comment]
A bug for submitting gap exercises is (re)introduced: unmodified starter code is submitted instead of actual submission file contents when modifying outside of gap. 

Please fix and test that modifying in or outside of the gap does not affect the submitted content within the gap.
  
There were also some trailing white-space, please run black for the updated files.

File: src/tidecli/api/routes.py [hunk 2]
```diff
@@ -197,3 +199,8 @@ def submit_task(
         raise click.ClickException("No feedback received")
 
     return TimFeedback(**feedback)
+
+
+def get_task_points(ide_task_id: str, doc_path: str) -> PointsData:
+    res = tim_request(endpoint=TASK_POINTS_ENDPOINT, method="GET", params={"ide_task_id": ide_task_id, "doc_path": doc_path})
+    return PointsData(**res)
```
Comment (sjsarsa) [inline src/tidecli/api/routes.py:206]: This throws a Pydantic error if the task doesn't give current points (no points specified or no submissions). Could be fixed on TIM side as well.

Comment (hannesi) [reply src/tidecli/api/routes.py:210]: Fixed by allowing None value for current points in cli

**Comment by hannesi** [comment]
> Please fix and test that modifying in or outside of the gap does not affect the submitted content within the gap.

Modifying the code outside of the gap currently prevents the answer from being submitted at all.

**Comment by hannesi** [comment]
Found a bug while creating task files for tasks that dont have pointsRule. Closing until its fixed.

File: src/tidecli/utils/file_handler.py (hunk 1 above)
Comment (sjsarsa) [reply src/tidecli/utils/file_handler.py:306]: Doesn't this reintroduce the bug as it was prior to the fix in main?
Also, the if-else should not be duplicated.

For reference, the fixed code in main is currently:

```
                    if validate_answer_file(answer_bycode, metadata_bycode):
                        # TODO: tarvitaan lisää testitapauksia,
                        # Validator OK
                        logger.info("Gap-type exercise answer file is valid.")
                        f1.content = "\n".join(answer_gapcode)
                    else:
                        logger.debug("Gap-type exercise answer not valid.")

                        # Validator complains about answer.
                        # Answer is submitted despite of complains.
                        f1.content = "\n".join(answer_gapcode)
                        # return []
```

**Review by sjsarsa — CHANGES_REQUESTED** [review]
Proposed gap exercise bug fix is not appropriate. Now it denies submission entirely if there are modifications outside the gap. Please fix and test again.

**Review by sjsarsa — COMMENTED** [review]
The bug is gone now, but it'd be good reintroduce the intended changes to get_task_data that were in the branch before the latest commit, i.e., the info level logging to debug level logging and formatting fixes.

**Pull request closed without merging**